data/posts/remix-jam-2025-recap.md

```diff
@@ -0,0 +1,279 @@
+
+#### Remixing the Server
+
+<iframe style="width:100%;aspect-ratio:16/9;" src="https://www.youtube.com/embed/dZbZgxWlzr8?si=4j3rkzJJOvarL8_Q" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
+
+After a break, Michael Jackson took the stage to talk about the backend pieces of Remix 3. Unlike Ryan's talk, just about everything Michael showed is already [open sourced](https://github.com/remix-run/remix), benchmarked, tested, and available on NPM.
+
+All the way back when working on Remix v1, it was a goal that Remix would be able to run inside of any JavaScript server runtime, which these days means: Node, Cloudflare Workers, Deno, Bun, Service Workers, etc.
+
+The goal was for Remix to be a full-stack JavaScript framework, but for Michael it always stopped a little short on the server side, and was only ever what he would consider "center stack".
+
+Seeing the server side of the new mountain they want to hike up, Michael deconstructs what a server is, what routes are, what handlers are, and puts all the pieces back together. He starts to build a JavaScript server from scratch, bringing in various packages as he goes:
+
+- [`@remix-run/route-pattern`](https://github.com/remix-run/remix/tree/main/packages/route-pattern)
+- [`@remix-run/node-fetch-server`](https://github.com/remix-run/remix/tree/main/packages/node-fetch-server)
+- [`@remix-run/form-data-parser`](https://github.com/remix-run/remix/tree/main/packages/form-data-parser)
+- [`@remix-run/multipart-parser`](https://github.com/remix-run/remix/tree/main/packages/multipart-parser)
+- [`@remix-run/file-storage`](https://github.com/remix-run/remix/tree/main/packages/file-storage)
+
+The bulk of the application is handled by the [`@remix-run/fetch-router`](https://github.com/remix-run/remix/tree/main/packages/fetch-router) package, an expressive, type-safe router built on the web Fetch API and `route-pattern`.
+
+### Bringing it all together (`hydrated()` and `Frame`)
+
+After showing how you can use these tools to easily build a simple CRUD (Create, Read, Update, Delete) application complete with file uploads, Michael switched to showing off the [bookstore app](https://github.com/remix-run/remix/tree/main/demos/bookstore) he and Ryan built that puts all these pieces together.
+
+It's definitely worth watching the full talk to see Michael walk through the code, as well as cracking it open yourself to see how all the various pieces work together. A simplified example of the core pieces looks like this:
+
+`fetch-router` requires named routes, separate from your route handlers. This is important for type-safety, code splitting, and static analysis (see the prior talks about Shopify Admin and Sidekick on why this is important.)
```

Comment:
Personally this feels like a lot of code for a recap post - we could just link to the bookstore demo in github if we wanted to reducer the code in here?  I think the best part of this section is hydrated/Frame which gets a bit buried beneath this code.

Reply:
Yeah, it is a lot of code. It's a hard balance, I want to hook people a bit with code, but definitely don't want to just recreate the whole example in here

I was able to reduce each of the blocks by like 30-50%, I think this is a fair compromise